### PR TITLE
submitting lesson03 - slicing lab

### PR DESCRIPTION
**Pull request by mxwllndrsn**

### CODE REVIEW

File: students/mxwllndrsn/lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#uw python 210
+#lesson 03
+#max anderson
+
+#list lab
+
+def fruits():
+
+    fruits = ['Apples', 'Pears', 'Oranges', 'Peaches']
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/list_lab.py]: Extra line - can be removed

File: students/mxwllndrsn/lesson03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    print(fruits)
+
+    print()
+    newfruit = input('name a fruit > ')
+
+    fruits.append(newfruit)
+    print(fruits)
+
+    print()
+    print('provide a number between 1 and', len(fruits), '>')
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/list_lab.py]: Nice use of len

File: students/mxwllndrsn/lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    print(fruits)
+
+    print()
+    print('only a few fruits begin with \'p\':' )
+
+    for fruit in fruits:
+        if fruit[0] == 'P' or fruit[0] == 'p':
+            print(fruit, end=' ')
+        else:
+            "no p fruits found"
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/list_lab.py]: Is this meant to be a print statement?

File: students/mxwllndrsn/lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+def fruits_2():
+
+    fruits = ['Apples', 'Pears', 'Oranges', 'Peaches']
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/list_lab.py]: Extra line can be removed.

File: students/mxwllndrsn/lesson03/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    try:
+        fruits.remove(preference)
+    except ValueError:
+        print('didn\'t find that one!')
+
+    print('got rid of it for you: ', fruits)
+
+def fruits_3():
+
+    fruits = ['Apples', 'Pears', 'Oranges', 'Peaches']
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/list_lab.py]: Extra line here

File: students/mxwllndrsn/lesson03/list_lab.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    for fruit in fruits[:]:
+        preference = input('Do you like {}? '.format(fruit.lower()))
+        while (preference != 'yes' and preference != 'no'):
+            preference = input('Please answer either \'yes\' or \'no\' >')
+        if preference == 'no':
+            fruits.remove(fruit)
+
+    print(fruits)
+
+def fruits_4():
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/list_lab.py]: Extra line here too.

File: students/mxwllndrsn/lesson03/slicing_lab.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
+#uw python 210
+#lesson 03
+#max anderson
+
+#slicing lab
+
+def switch_firstlast(seq):
+
+    return seq[-1:] + seq[1:len(seq)-1] + seq[0:1]
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/slicing_lab.py]: Delete extra line in this function as well as following ones.

File: students/mxwllndrsn/lesson03/slicing_lab.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    assert remove_alternating(atuple) == (2, 33, 23, 99)
+
+    assert alternating_mid(astring) == ' sas'
+    assert alternating_mid(atuple) ==  ()
+
+    assert reverses(astring) == 'gnirts a si siht'
+    assert reverses(atuple) == (123, 99, 77, 23, 66, 33, 55, 2)
+
+    assert third_order(astring) == 'is a stringthis '
+    assert third_order(atuple) == (33, 66, 23, 77, 99, 123, 2, 55)
```
Comment (charlesrobison) [inline students/mxwllndrsn/lesson03/slicing_lab.py]: Good set of tests!

File: students/mxwllndrsn/lesson03/list_lab.py (hunk 1 above)
Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/list_lab.py]: Charles,
This list does not exist globally when ran or imported in iPython. Not sure what the deal is but needs to exist in each function for this to work.

File: students/mxwllndrsn/lesson03/list_lab.py (hunk 3 above)
Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/list_lab.py]: yes, should be print("no p fruits found")

File: students/mxwllndrsn/lesson03/list_lab.py (hunk 4 above)
Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/list_lab.py]: same as above

File: students/mxwllndrsn/lesson03/list_lab.py (hunk 5 above)
Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/list_lab.py]: same as above

File: students/mxwllndrsn/lesson03/list_lab.py (hunk 6 above)
Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/list_lab.py]: Do you mean the function def or the print statement?

File: students/mxwllndrsn/lesson03/slicing_lab.py (hunk 7 above)
Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/slicing_lab.py]: not sure what you mean

File: students/mxwllndrsn/lesson03/list_lab.py (hunk 1 above)
Comment (natasha-aleksandrova) [reply students/mxwllndrsn/lesson03/list_lab.py]: Charles was referring to extra line on line 8. (pep8 formatting).

Comment (mxwllndrsn) [reply students/mxwllndrsn/lesson03/list_lab.py]: Oh, I will be sure to implement this in future. I feel it more readable with space after def but if that's the standard for the course I will use it. (don't think i used it on strformat_lab though)

Do i need to change it to receive grade?

**Pull request merged**